Feature/test

nh3 opened the pull request.

nh3 commented on .travis.yml:
```diff
@@ -0,0 +1,46 @@
+    - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION
+    - source activate test-environment
+    - conda install scanpy
+    - pip install -U --upgrade-strategy only-if-needed scanpy
```
It's to install the latest version of scanpy (1.3.2). I have local feature branches for pca and tsne that works with scanpy-1.3.1. But I still think v1.3.2 is much better than v1.3.1. Perhaps the better solution is to have bioconda roll out scanpy v1.3.2.

pinin4fjords replied:
As I mentioned, if possible we'd like to stick with the Conda versions, due to our currently agreed strategy and the impact it has on containers etc. At least that will be the minimum product. We can always push for upgrade of the conda package and use it later.

If there's something completely broken in the Conda version, let us know and we'll have to deal with it. But provided the fundamental workflow is possible (with whatever intermediate formats), please just use the Conda version.

We may need to change strategy and abandon Conda (it's behind the curve on R stuff too), but that's a separate conversation, and something we would need to agree as a group.

Ping @pcm32 for FYI.

pinin4fjords replied:
Having checked, I note that 1.3.1 is only from September. 1.3.2 is only 11 days old! I don't think we can be too irritated at Bioconda on that basis. Just keep your 1.3.2. code somewhere safe, and we can upgrade when Bioconda moves, which doesn't look like it should be too long.

pinin4fjords commented on .travis.yml:
```diff
@@ -0,0 +1,46 @@
+
+      conda config --add channels defaults;
+      conda config --add channels conda-forge;
+      conda config --add channels bioconda;
```
So, you need to use an environment creation step here- as per https://github.com/ebi-gene-expression-group/r-seurat-scripts/blob/master/.travis.yml. Have the Scanpy install happen there instead of where you have it now. Then it will be part of the caching setup.

pinin4fjords commented:
Hi @nh3 - looking pretty good (aside from the Conda issues etc). I think a comment disappeared in the revisions so I'll re-iterate here: 1.3.2 is only 11 days old, so I don't think we can worry too much about the Conda repos being out of date just yet.

Could you also add some descriptive text to this PR's description, for posterity? Describe what the PR contains and what the overall objective is. Thank you!

nh3 commented:
Yes, I currently have branches for both 1.3.1 and 1.3.2.

Time-wise the two release are very close. The changes are quite substantial, though. As far as I've encountered, v1.3.2 has refactored much of the code for plotting, cleaned out some inconsistence and added about 40% more options to plotting functions for better control. These changes are probably not essential for current purpose. The annoying thing is that, the API documentation is only for v1.3.2. I have to check the actual code to find out which options were not present in v1.3.1...

﻿On 16/10/2018, 10:42, "Jonathan Manning" <notifications@github.com> wrote:

    @pinin4fjords commented on this pull request.
    ________________________________________
    In 
    .travis.yml <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ebi-2Dgene-2Dexpression-2Dgroup_scanpy-2Dscripts_pull_8-23discussion-5Fr225468980&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=Gqrgrpvmsq-v8t6_Lt2aYdQdvBpIuPJgAyozOxDjHHU&m=CQ6sP7fBZmLjbP__LMwawu7bFCpdHnFl_NLcsAI2FAc&s=37JkmpUqrhFz51ejuJwRI3_CaqV6lyQxOsB6YiKcf8Y&e=>:
    > +      conda config --set always_yes yes --set changeps1 no;
    +      conda update -q conda;
    +
    +      conda info -a;
    +
    +      conda config --add channels defaults;
    +      conda config --add channels conda-forge;
    +      conda config --add channels bioconda;
    +    fi
    +
    +install:
    +    - export PATH="$HOME/miniconda/bin:$PATH";
    +    - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION
    +    - source activate test-environment
    +    - conda install scanpy
    +    - pip install -U --upgrade-strategy only-if-needed scanpy
    Having checked, I note that 1.3.1 is only from September. 1.3.2 is only 11 days old! I don't think we can be too irritated at Bioconda on that basis. Just keep your 1.3.2. code somewhere safe, and we can upgrade when Bioconda moves, which doesn't look like
     it should be too long.
    —
    You are receiving this because you authored the thread.
    Reply to this email directly, 
    view it on GitHub <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ebi-2Dgene-2Dexpression-2Dgroup_scanpy-2Dscripts_pull_8-23discussion-5Fr225468980&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=Gqrgrpvmsq-v8t6_Lt2aYdQdvBpIuPJgAyozOxDjHHU&m=CQ6sP7fBZmLjbP__LMwawu7bFCpdHnFl_NLcsAI2FAc&s=37JkmpUqrhFz51ejuJwRI3_CaqV6lyQxOsB6YiKcf8Y&e=>, or 
    mute the thread <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AAZuk03hblYerw4Pw56UzUcfHbSrXAJ9ks5ulaoegaJpZM4Xd6D9&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=Gqrgrpvmsq-v8t6_Lt2aYdQdvBpIuPJgAyozOxDjHHU&m=CQ6sP7fBZmLjbP__LMwawu7bFCpdHnFl_NLcsAI2FAc&s=62saG7GeZA4xElc9IaOzzY5Zxy_OjjQkmfSn_4iSa40&e=>.
    {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/ebi-gene-expression-group/scanpy-scripts","title":"ebi-gene-expression-group/scanpy-scripts","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/ebi-gene-expression-group/scanpy-scripts"}},"updates":{"snippets":[{"icon":"PERSON","message":"@pinin4fjords commented on #8"}],"action":{"name":"View Pull Request","url":"https://github.com/ebi-gene-expression-group/scanpy-scripts/pull/8#discussion_r225468980"}}}[
    {
    "@context": "http://schema.org",
    "@type": "EmailMessage",
    "potentialAction": {
    "@type": "ViewAction",
    "target": "https://github.com/ebi-gene-expression-group/scanpy-scripts/pull/8#discussion_r225468980",
    "url": "https://github.com/ebi-gene-expression-group/scanpy-scripts/pull/8#discussion_r225468980",
    "name": "View Pull Request"
    },
    "description": "View this Pull Request on GitHub",
    "publisher": {
    "@type": "Organization",
    "name": "GitHub",
    "url": "https://github.com"
    }
    },
    {
    "@type": "MessageCard",
    "@context": "http://schema.org/extensions",
    "hideOriginalBody": "false",
    "originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
    "title": "@pinin4fjords commented on 8",
    "sections": [
    {
    "text": "",
    "activityTitle": "**Jonathan Manning**",
    "activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
    "activitySubtitle": "@pinin4fjords",
    "facts": [
    
    ]
    }
    ],
    "potentialAction": [
    {
    "targets": [
    {
    "os": "default",
    "uri": "https://github.com/ebi-gene-expression-group/scanpy-scripts/pull/8#discussion_r225468980"
    }
    ],
    "@type": "OpenUri",
    "name": "View on GitHub"
    },
    {
    "name": "Unsubscribe",
    "@type": "HttpPOST",
    "target": "https://api.github.com",
    "body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 393715965\n}"
    }
    ],
    "themeColor": "26292E"
    }
    ]




-- 
 The Wellcome Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.

nh3 commented:
This PR somehow included scripts not yet in development. I'll create a new one with only the test scripts.
